lib/jsonapi/utils/response/formatters.rb

```diff
@@ -96,9 +106,18 @@ def build_response_document(object, options)
 
           if object.respond_to?(:to_ary)
             records = build_collection(object, options)
-            results.add_result(JSONAPI::ResourcesOperationResult.new(:ok, records, result_options(object, options)))
+
+            if options[:source_resource].present? && options[:relationship_type].present?
+              results.add_result(JSONAPI::RelatedResourcesOperationResult.new(:ok,
+                                                          options[:source_resource],
```

Comment:
As it already [happens on formatters](https://github.com/tiagopog/jsonapi-utils/blob/master/lib/jsonapi/utils/response/formatters.rb#L170), you could try inferring some stuff from the params found in the `@request` object rather than making both `:source_resource` and `: relationship_type ` mandatory in order to achieve such a feature.

I was also thinking on changing a little bit the way the optionals are passed to the helper method to something like this:

```ruby
# Usage 1:
# - Source resource is inferred: @request.source_klass #=> UserResource
# - Relationship is inferred: params[:relationship].to_sym || @request.resource_klass._type #=> :posts
jsonapi_render json: @posts, options: { source: @user }

# Usage 2:
# - Source resource is defined via optionals
# - Relationship is inferred: params[:relationship].to_sym || @request.resource_klass._type #=> :posts
jsonapi_render json: @posts, options: { source: WeirdNamespace::UserResource.new(@user, context) }

# Usage 3:
# - Source resource is inferred: @request.source_klass #=> UserResource
# - Relationship is defined via optionals
jsonapi_render json: @posts, options: { source: @user, relationship: :posts }

# Usage 4:
# - Source resource is defined via optionals
# - Relationship is defined via optionals
jsonapi_render json: @posts, options: { source: WeirdNamespace::UserResource.new(@user, context), relationship: :posts }
```